codec-sockjs/src/main/java/io/netty/handler/codec/sockjs/util/JsonConverter.java

```diff
@@ -0,0 +1,275 @@
+ * Copyright 2014 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.netty.handler.codec.sockjs.util;
+
+import com.fasterxml.jackson.core.JsonGenerationException;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.io.CharTypes;
+import com.fasterxml.jackson.core.json.JsonWriteContext;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.handler.codec.http.websocketx.TextWebSocketFrame;
+import io.netty.util.CharsetUtil;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public final class JsonConverter {
+
+    private static final ObjectMapper MAPPER = createObjectMapper();
+    private static final String[] EMPTY_STRING_ARRAY = {};
+    private static final char[] HEX_CHARS = CharTypes.copyHexChars();
+    private static final int[] ESCAPE_CODES = CharTypes.get7BitOutputEscapes();
+
+    private static ObjectMapper createObjectMapper() {
+        return new ObjectMapper()
+                .registerModule(new SimpleModule("netty-codec-sockjs")
+                .addSerializer(String.class, new StringSerializer()));
+    }
+
+    private JsonConverter() {
+    }
+
+    @SuppressWarnings("resource")
+    public static String[] decode(final TextWebSocketFrame frame) throws IOException {
+        final ByteBuf content = frame.content();
+        if (content.readableBytes() == 0) {
+            return EMPTY_STRING_ARRAY;
+        }
+        final ByteBufInputStream byteBufInputStream = new ByteBufInputStream(content);
+        try {
+            final byte firstByte = content.getByte(0);
+            if (firstByte == '[') {
+                return MAPPER.readValue(byteBufInputStream, String[].class);
+            } else if (firstByte == '{') {
+                return new String[]{content.toString(CharsetUtil.UTF_8)};
+            } else {
+                return new String[]{MAPPER.readValue(byteBufInputStream, String.class)};
+            }
+        } finally {
+            if (byteBufInputStream != null) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "true" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2583)